analyzer/src/main/kotlin/managers/MavenDependencyHandler.kt

```diff
@@ -55,10 +56,16 @@ class MavenDependencyHandler(
      */
     val sbtMode: Boolean
 ) : DependencyHandler<DependencyNode> {
-    override fun identifierFor(dependency: DependencyNode): String {
+    override fun identifierFor(dependency: DependencyNode): Identifier {
         val id = dependency.identifier()
-        val type = if (isLocalProject(id)) managerName else "Maven"
-        return "$type:$id"
+        val parts = id.split(':')
+
+        return Identifier(
+            type = if (isLocalProject(id)) managerName else "Maven",
+            namespace = parts.getOrElse(0) { "" },
+            name = parts.getOrElse(1) { "" },
+            version = parts.getOrElse(2) { "" },
+        )
```

Comment:
Sorry, looking at this again, there's no need to concat and split the `id` again, but we can just use

    override fun identifierFor(dependency: DependencyNode): Identifier {
        val id = dependency.identifier()

        return Identifier(
            type = if (isLocalProject(id)) managerName else "Maven",
            namespace = dependency.artifact.groupId,
            name = dependency.artifact.artifactId,
            version = dependency.artifact.version
        )
    }

Reply:
As a follow-up, I'd personally also inline the two uses of the `DependencyNode.identifier()` convenience function.

Reply:
I'll add your above suggestion to my follow-up PR.